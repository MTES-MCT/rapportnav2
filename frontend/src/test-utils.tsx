import React, { ReactElement } from 'react'
import { render, renderHook, RenderOptions } from '@testing-library/react'
import UIThemeWrapper from './ui/ui-theme-wrapper'
import { BrowserRouter } from 'react-router-dom'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <UIThemeWrapper>
      <BrowserRouter>{children}</BrowserRouter>
    </UIThemeWrapper>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

const customRenderHook = (hook: any, options?: Omit<RenderOptions, 'wrapper'>) =>
  renderHook(() => hook(), { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
export { customRenderHook as renderHook }
