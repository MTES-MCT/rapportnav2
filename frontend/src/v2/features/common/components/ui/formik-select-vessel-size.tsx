import { FormikSelect, FormikSelectProps } from '@mtes-mct/monitor-ui'
import styled from 'styled-components'
import { useVessel } from '../../hooks/use-vessel'

export const FormikSelectVesselSize = styled((props: Omit<FormikSelectProps, 'options'>) => {
  const { vesselSizeOptions } = useVessel()
  return (
    <FormikSelect isLight={true} isRequired={true} isErrorMessageHidden={true} {...props} options={vesselSizeOptions} />
  )
})({})
