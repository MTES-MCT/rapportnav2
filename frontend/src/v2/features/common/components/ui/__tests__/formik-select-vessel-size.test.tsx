import { Formik } from 'formik'
import { vi } from 'vitest'
import { render } from '../../../../../../test-utils'
import { FormikSelectVesselSize } from '../formik-select-vessel-size'

const handleSubmit = vi.fn()
describe('FormikSelectVesselSize', () => {
  it('should match the snapshot', () => {
    const wrapper = render(
      <Formik initialValues={{ vesselSize: '' }} onSubmit={handleSubmit}>
        <FormikSelectVesselSize name="vesselSize" label="select vessel Size" />
      </Formik>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
