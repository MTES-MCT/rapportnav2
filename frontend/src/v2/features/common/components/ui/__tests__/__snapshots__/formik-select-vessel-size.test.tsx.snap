// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormikSelectVesselSize > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gtlvkJ cuIUGk Element-Field Field-Select sc-btRGtK"
      >
        <label
          class="sc-loYsQo iILbqD Element-Label"
          for="vesselSize"
        >
          select vessel Size
        </label>
        <div
          class="sc-kHpflN iSHmec"
        >
          <div
            class="rs-picker rs-picker-select rs-picker-default rs-picker-toggle-wrapper rs-picker-cleanable"
          >
            <div
              aria-controls="vesselSize-listbox"
              aria-describedby="vesselSize-describe"
              aria-disabled="false"
              aria-expanded="false"
              aria-haspopup="listbox"
              class="rs-picker-toggle rs-btn rs-btn-default"
              id="vesselSize"
              role="combobox"
              tabindex="0"
            >
              <div
                class="rs-stack"
                style="align-items: center;"
              >
                <div
                  class="rs-stack-item"
                  style="flex-grow: 1; overflow: hidden;"
                >
                  <input
                    aria-hidden="true"
                    class="rs-picker-toggle-textbox rs-picker-toggle-read-only"
                    data-testid="picker-toggle-input"
                    name="vesselSize"
                    readonly=""
                    style="pointer-events: none;"
                    tabindex="-1"
                    value=""
                  />
                  <span
                    class="rs-picker-toggle-placeholder"
                    data-testid="picker-describe"
                    id="vesselSize-describe"
                  >
                    Sélectionner
                  </span>
                </div>
                <div
                  class="rs-stack-item rs-picker-toggle-indicator"
                >
                  <svg
                    aria-hidden="true"
                    aria-label="angle down"
                    class="rs-picker-caret-icon rs-icon rs-icon"
                    data-category=""
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="0 0 21 32"
                    width="1em"
                  >
                    <path
                      d="M19.196 13.143a.612.612 0 01-.179.411l-8.321 8.321c-.107.107-.268.179-.411.179s-.304-.071-.411-.179l-8.321-8.321c-.107-.107-.179-.268-.179-.411s.071-.304.179-.411l.893-.893a.582.582 0 01.411-.179c.143 0 .304.071.411.179l7.018 7.018 7.018-7.018c.107-.107.268-.179.411-.179s.304.071.411.179l.893.893a.617.617 0 01.179.411z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="rs-toast-provider"
    >
      <div
        class="rs-toast-container rs-toast-container-top-center"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-center"
      />
      <div
        class="rs-toast-container rs-toast-container-top-start"
      />
      <div
        class="rs-toast-container rs-toast-container-top-end"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-start"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-end"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gtlvkJ cuIUGk Element-Field Field-Select sc-btRGtK"
    >
      <label
        class="sc-loYsQo iILbqD Element-Label"
        for="vesselSize"
      >
        select vessel Size
      </label>
      <div
        class="sc-kHpflN iSHmec"
      >
        <div
          class="rs-picker rs-picker-select rs-picker-default rs-picker-toggle-wrapper rs-picker-cleanable"
        >
          <div
            aria-controls="vesselSize-listbox"
            aria-describedby="vesselSize-describe"
            aria-disabled="false"
            aria-expanded="false"
            aria-haspopup="listbox"
            class="rs-picker-toggle rs-btn rs-btn-default"
            id="vesselSize"
            role="combobox"
            tabindex="0"
          >
            <div
              class="rs-stack"
              style="align-items: center;"
            >
              <div
                class="rs-stack-item"
                style="flex-grow: 1; overflow: hidden;"
              >
                <input
                  aria-hidden="true"
                  class="rs-picker-toggle-textbox rs-picker-toggle-read-only"
                  data-testid="picker-toggle-input"
                  name="vesselSize"
                  readonly=""
                  style="pointer-events: none;"
                  tabindex="-1"
                  value=""
                />
                <span
                  class="rs-picker-toggle-placeholder"
                  data-testid="picker-describe"
                  id="vesselSize-describe"
                >
                  Sélectionner
                </span>
              </div>
              <div
                class="rs-stack-item rs-picker-toggle-indicator"
              >
                <svg
                  aria-hidden="true"
                  aria-label="angle down"
                  class="rs-picker-caret-icon rs-icon rs-icon"
                  data-category=""
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="0 0 21 32"
                  width="1em"
                >
                  <path
                    d="M19.196 13.143a.612.612 0 01-.179.411l-8.321 8.321c-.107.107-.268.179-.411.179s-.304-.071-.411-.179l-8.321-8.321c-.107-.107-.179-.268-.179-.411s.071-.304.179-.411l.893-.893a.582.582 0 01.411-.179c.143 0 .304.071.411.179l7.018 7.018 7.018-7.018c.107-.107.268-.179.411-.179s.304.071.411.179l.893.893a.617.617 0 01.179.411z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
